core-common/src/main/java/org/mapstruct/MappingConstants.java

```diff
@@ -0,0 +1,45 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct;
+
+/**
+ *
+ * @author Sjaak Derksen
+ */
+public final class MappingConstants {
+
+    private MappingConstants() {
+    }
+
+    /**
+     * Represents a {@code null} source or target.
+     */
+    public static final String NULL = "<NULL>";
+
+    /**
+     * Represents any source that is not already mapped by either a defined mapping or by means of name based mapping.
+     */
+    public static final String ANY = "<ANY>";
```

Comment:
> Should we name it "<ANY_REMAINING>" or so? I am having the same feeling as you did with the "", i.e. it can be understood to cover *anything? It's cool with me, if you think there is no issue.

The problem I have is that <ANY_UNMATCHED> is actually a larger set than <ANY>.. That feels strange.. I opt for <ANY_REMAINING> ( :+1: @gunnarmorling ). If there are no further objections ( @agudian ), or forever hold your peace.. :Smile: